op-e2e/e2eutils/disputegame/helper.go

```diff
@@ -172,14 +174,26 @@ func (h *FactoryHelper) StartCannonGameWithCorrectRoot(ctx context.Context, roll
 		L2Claim:       challengedOutput.OutputRoot,
 		L2BlockNumber: challengedOutput.L2BlockNumber,
 	}
-	provider := cannon.NewTraceProviderFromInputs(testlog.Logger(h.t, log.LvlInfo).New("role", "CorrectTrace"), metrics.NoopMetrics, cfg, inputs, cfg.Datadir)
-	rootClaim, err := provider.Get(ctx, math.MaxUint64)
+
+	maxDepth := uint64(math.MaxUint64)
+	provider := cannon.NewTraceProviderFromInputs(
+		testlog.Logger(h.t, log.LvlInfo).New("role", "CorrectTrace"),
+		metrics.NoopMetrics,
+		cfg,
+		inputs,
+		cfg.Datadir,
+		maxDepth,
+	)
+	pos := faultTypes.NewPosition(int(maxDepth), int(maxDepth))
```

Comment:
You actually want `faultTypes.NewPositionFromGIndex(1)` here - it's the root position that we need the claim for. Might be worth defining the root position as a constant actually.